modules/mongodb/templates/mongodb-backup-s3.sh.erb

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+
+BACKUP_NODE=<%= @backup_node %>
+BACKUP_DIR=<%= @backup_dir %>
+BACKUP_FILE=mongodump-<%= @fqdn %>.$(date +%F_%T).tar.gz.gpg
+KEY_FINGERPRINT=<%= @private_gpg_key_fingerprint %>
+S3_BUCKET=<%= @s3_bucket %>
+
+
+exec 1> >(/usr/bin/logger -s -t $(basename $0)) 2>&1
+
+function time_taken {
+  TIME="$(date +%s%N)"
+  $@
+  TIME="$(($(date +%s%N)-TIME))"
+  SECS="$((TIME/1000000000))"
+  MILLISECS="$((TIME/100000000))"
+  printf " FINISHED IN: %02d:%02d:%02d.%02d\n" "$((SECS/3600%24))" "$((SECS/60%60))" "$((SECS%60))" "${MILLISECS}"
+
+}
+
+echo "................STARTING BACKUP..................."
```

Comment:
Is the mongodump output silent? If so I'd be inclined to agree that even a "Backup started at $(date)" and "Backup finished at $(date)" would be useful, but we just need it for syslog so the padding on the output (the fullstops) isn't needed.